lib/compiler/passes/generate-bytecode.js

```diff
@@ -425,15 +435,25 @@ function generateBytecode(ast) {
         buildElementsCode(node.elements, {
           sp: context.sp + 1,
           env: context.env,
+          pluck: [],
           action: context.action
         })
       );
     },
 
     labeled(node, context) {
-      const env = cloneEnv(context.env);
+      let env = context.env;
+      const label = node.label;
+      const sp = context.sp + 1;
+
+      if (label) {
+        env = cloneEnv(context.env);
+        context.env[node.label] = sp;
+      }
 
-      context.env[node.label] = context.sp + 1;
+      if (node.pick && context.pluck) {
```

Comment:
No need to check properties that have to exist:
```suggestion
      if (node.pick) {
```